src/main/kotlin/no/nav/faktureringskomponenten/service/beregning/AntallBeregner.kt

```diff
@@ -0,0 +1,64 @@
+package no.nav.faktureringskomponenten.service.beregning
+
+import mu.KotlinLogging
+import java.math.BigDecimal
+import java.math.RoundingMode
+import java.time.LocalDate
+import java.time.temporal.ChronoUnit
+
+private val log = KotlinLogging.logger { }
+
+class AntallBeregner {
+    companion object {
+        fun antallForPeriode(fom: LocalDate, tom: LocalDate): BigDecimal {
+            val gjelderSammeMånedOgÅr = fom.year == tom.year && fom.monthValue == tom.monthValue
+            if (gjelderSammeMånedOgÅr) {
+                log.debug("Beregner antall for periode $fom og $tom på én og samme måned")
+                return beregnAntallForEnMåned(fom, tom)
+            }
+            log.debug("Beregner antall for periode $fom og $tom på flere måneder")
+            return beregnAntallForFlereMåneder(fom, tom)
+        }
+
+        private fun beregnAntallForFlereMåneder(fom: LocalDate, tom: LocalDate): BigDecimal {
+            var totalAngittAntall = BigDecimal.ZERO
+            var currentDate = fom
+
+            while (currentDate <= tom) {
+                val erFørsteMåned = fom == currentDate
+                val erSisteMåned = currentDate.year == tom.year && currentDate.month == tom.month
+                val totalAntallDagerForMåned = currentDate.lengthOfMonth()
+                val dagerIgjenAvMåneden = totalAntallDagerForMåned - (currentDate.dayOfMonth - 1)
+
+                val angittAntallForMåned =
+                    BigDecimal(dagerIgjenAvMåneden.toDouble() / totalAntallDagerForMåned.toDouble())
+                        .setScale(2, RoundingMode.HALF_UP)
+
+                when {
+                    erFørsteMåned -> {
+                        totalAngittAntall = angittAntallForMåned
+                        currentDate = currentDate.withDayOfMonth(1)
+                    }
+
+                    erSisteMåned -> {
+                        val daysInMonth = tom.dayOfMonth - (currentDate.dayOfMonth - 1)
+                        totalAngittAntall += BigDecimal(daysInMonth.toDouble() / totalAntallDagerForMåned.toDouble())
+                            .setScale(2, RoundingMode.HALF_UP)
+                    }
+
+                    else -> {
+                        totalAngittAntall += angittAntallForMåned
+                    }
+                }
+                currentDate = currentDate.plusMonths(1)
```

Comment:
Det gir vel ikke mening at currentDate plutselig er en måned frem i tid? Burde denne variabelen hete noe annet?

Reply:
Tror det er vanlig å bruke "current" for en variabel i en loop, for å beskrive hvilken verdi man bruker i iterasjonen. Har foreslått kode uten loops thou, som gjør det etter min mening en del mer lesbart.

Reply:
tempDato kanskje?

Reply:
Hva er det datoen representerer?

Reply:
Ikke lengre relevant